File: frontend/edenBooks/src/modules/carriers/services/carrierService.ts
import { carrierApi } from "../api/carrierApi";
import { ICarrier } from "../interfaces/ICarrier";
import { ICarrierService } from "../interfaces/ICarrierService";

export const CarrierService: ICarrierService = {
  CreateCarrier: async (carrier: ICarrier) =>carrierApi.CreateCarrier(carrier),
  GetCarriers: async () => carrierApi.GetCarriers(),
  GetCarrierById: async (id: number) => carrierApi.GetCarrierById(id),
  UpdateCarrier: async (id: number, carrier: ICarrier) =>carrierApi.UpdateCarrier(id, carrier),
  DeleteCarrier: async (id: number) => carrierApi.DeleteCarrier(id),
};

File: backend/internal/modules/carriers/domain/ports/services/carrier_service.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/carriers/domain/entities"
)

type CarrierService interface {
	CreateCarrier(c *entities.Carrier) error

	UpdateCarrier(c *entities.Carrier) error

	DeleteCarrier(id uint) error

	GetCarrierByID(id uint) (*entities.Carrier, error)

	GetFilteredCarrieres(filters map[string]string) ([]entities.Carrier, error)
}
